spacy_curated_transformers/pipeline/transformer.py

```diff
@@ -141,14 +143,18 @@ def listening_components(self) -> List[str]:
         """
         return list(self.listener_map.keys())
 
-    def add_listener(self, listener: TransformerListener, component_name: str) -> None:
+    def add_listener(
+        self, listener: TransformerListenerModelT, component_name: str
+    ) -> None:
         """Add a listener for a downstream component. Usually internals."""
+        assert TransformerListener.is_listener(listener)
```

Comment:
This is on a boundary with downstream user, we should use a regular exception here.

Reply:
Fixed.